core/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -568,6 +568,18 @@ public Object readGenericValue() throws IOException {
     }
 
     /**
+     * Reads an optional {@link NamedWriteable} from the current stream.
+     * */
+    public <C> C readOptionalNamedWriteable() throws IOException {
+        boolean readable = readBoolean();
+        if (readable) {
```

Comment:
super minor thing, but here we could do if (readBoolean()) directly  without the readable variable